OpenRA.Mods.Common/Effects/Beacon.cs

```diff
@@ -96,6 +96,9 @@ void IEffect.Tick(World world)
 
 			if (clock != null)
 				clock.Tick();
+
+			if (duration > 0 && duration < tick++)
```

Comment:
You'll want to use either `++tick` or `<=` to avoid staying for 1 tick longer than asked for.